### PR TITLE
Add Gov/Distribution precompiles

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Gov precompile supports:
- vote
- deposit

Distribution precompile supports:
- set withdraw address (subject to distribution module param control)
- withdraw delegation rewards

## Testing performed to validate your change
unit tests/local sei test

### CODE REVIEW

File: precompiles/common/expected_keepers.go [hunk 1]
```diff
@@ -43,3 +44,13 @@ type StakingKeeper interface {
...
 }
+
+type GovKeeper interface {
+	AddVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress, options govtypes.WeightedVoteOptions) error
+	AddDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress, depositAmount sdk.Coins) (bool, error)
+}
+
+type DistributionKeeper interface {
+	SetWithdrawAddr(ctx sdk.Context, delegatorAddr sdk.AccAddress, withdrawAddr sdk.AccAddress) error
+	WithdrawDelegationRewards(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) (sdk.Coins, error)
```
Comment (philipsu522) [inline precompiles/common/expected_keepers.go:55]: do we want to support 
- `withdraw-all-rewards` 
- `set-withdraw-addr`
as well?

File: precompiles/gov/gov.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	pcommon "github.com/sei-protocol/sei-chain/precompiles/common"
+)
+
+const (
+	VoteMethod    = "vote"
+	DepositMethod = "deposit"
+)
+
+const (
+	GovAddress = "0x0000000000000000000000000000000000001006"
```
Comment (philipsu522) [inline precompiles/gov/gov.go:22]: nit - is there a way to ensure that we don't reuse an addr for precompile? maybe we should add some precompile registry (or something like in app.go) and have a unit test

File: precompiles/common/expected_keepers.go [hunk 3]
```diff
@@ -43,3 +44,13 @@ type StakingKeeper interface {
 	BeginRedelegate(goCtx context.Context, msg *stakingtypes.MsgBeginRedelegate) (*stakingtypes.MsgBeginRedelegateResponse, error)
 	Undelegate(goCtx context.Context, msg *stakingtypes.MsgUndelegate) (*stakingtypes.MsgUndelegateResponse, error)
 }
+
+type GovKeeper interface {
+	AddVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress, options govtypes.WeightedVoteOptions) error
```
Comment (philipsu522) [inline precompiles/common/expected_keepers.go:49]: do we want to add `submit-proposal`?

**Review by philipsu522 — APPROVED** [review]

File: precompiles/common/expected_keepers.go (hunk 1 above)
Comment (codchen) [reply precompiles/common/expected_keepers.go:55]: i think those are the two currently implemented, or are you referring to smth else?

File: precompiles/gov/gov.go (hunk 2 above)
Comment (codchen) [reply precompiles/gov/gov.go:22]: yeah i can add a check in the setup code

File: precompiles/common/expected_keepers.go (hunk 3 above)
Comment (codchen) [reply precompiles/common/expected_keepers.go:49]: do we currently allow users to submit proposal on UI on cosmos side? (feel like this precompile is only relevant for users interacting with the chain through evm UI)

**Pull request merged**